receiver/couchbasereceiver/client_test.go

```diff
@@ -0,0 +1,333 @@
+					w.Write(data)
+				}))
+				defer ts.Close()
+
+				tc := createTestClient(t, ts.URL)
+
+				// Load the valid data into a struct to compare
+				var expected []*bucket
+				err := json.Unmarshal(data, &expected)
+				require.NoError(t, err)
+
+				buckets, err := tc.GetBuckets(context.Background(), "/path/to/bucket_info")
+				require.NoError(t, err)
+				require.Equal(t, expected, buckets)
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.desc, tc.testFunc)
+	}
+}
+
+func TestGetBucketStats(t *testing.T) {
+
+	testCases := []struct {
+		desc     string
+		testFunc func(*testing.T)
+	}{
+		{
+			desc: "Non-200 Response",
+			testFunc: func(t *testing.T) {
+				// Setup test server
+				ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+					w.WriteHeader(http.StatusUnauthorized)
+				}))
+				defer ts.Close()
+
+				tc := createTestClient(t, ts.URL)
+
+				stats, err := tc.GetBucketStats(context.Background(), "/path/to/stats")
+				require.Nil(t, stats)
+				require.EqualError(t, err, "non 200 code returned 401")
+			},
+		},
+		{
+			desc: "Bad payload returned",
+			testFunc: func(t *testing.T) {
+				// Setup test server
+				ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+					w.Write([]byte("[]"))
+				}))
+				defer ts.Close()
+
+				tc := createTestClient(t, ts.URL)
+
+				stats, err := tc.GetBucketStats(context.Background(), "/path/to/stats")
+				require.Nil(t, stats)
+				require.Contains(t, err.Error(), "failed to decode response payload")
+			},
+		},
+		{
+			desc: "Successful call",
+			testFunc: func(t *testing.T) {
+				data := loadAPIResponseData(t, bucketStatsAPIResponseFile)
+
+				// Setup test server
+				ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+					w.Write(data)
+				}))
+				defer ts.Close()
+
+				tc := createTestClient(t, ts.URL)
+
+				// Load the valid data into a struct to compare
+				var expected bucketStats
+				err := json.Unmarshal(data, &expected)
+				require.NoError(t, err)
+
+				stats, err := tc.GetBucketStats(context.Background(), "/path/to/stats")
+				require.NoError(t, err)
+				require.Equal(t, &expected, stats)
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.desc, tc.testFunc)
+	}
+}
+
+func createTestClient(t *testing.T, baseEndpoint string) client {
+	t.Helper()
+	cfg := createDefaultConfig().(*Config)
+	cfg.Endpoint = baseEndpoint
+
+	testClient, err := newClient(cfg, componenttest.NewNopHost(), zap.NewNop())
+	require.NoError(t, err)
+	return testClient
+}
+
+func loadAPIResponseData(t *testing.T, fileName string) []byte {
+	t.Helper()
+
+	file, err := os.Open(filepath.Join("testdata", "apiresponses", fileName))
+	require.NoError(t, err)
+
+	defer func() {
+		closeErr := file.Close()
+		require.NoError(t, closeErr)
+	}()
+
+	data, err := io.ReadAll(file)
+	require.NoError(t, err)
+
+	return data
+}
```

Comment:
We haven't been following a standard on this, but perhaps we should. `ioutil.ReadFile` is much more concise and handles the edge cases. Do you think it's worth standardizing on that so we don't have to worry about this level of detail in each scraper?

Reply:
I don't have strong feelings one way or another. Since this is just a test helper instead of production and a very controlled environment with what it's loading I'm not too concerned about edge cases but also I'm find with just using `ioutil.ReadFile` if that's what we're doing elsewhere in the code base. I'll update.